### PR TITLE
Feature 1374 python packages

### LINKED ISSUE #1374: Add recording of Python packages to Adding Use Cases documentation
## Describe the Task ##
This issue arose from a request from EMC to compile a complete list of all Python packages that are used for the METplus wrappers use cases, METcalcpy, and METplotpy.  To run the METplus wrappers, the only Python package that is required besides the standard packages is dateutil (python-dateutil). However, some METplus use cases have additional Python package dependencies to run.  Use cases that utilize Python Embedding in the MET tools need to have xarray, numpy, pandas, and netCDF4, as well as any other Python packages that are used in the script that is used. The full list of package dependencies for the use cases depends on which cases need to be supported on WCOSS2, and it was determined that EMC users would like to have all of the use cases available for running on WCOSS2.  

The required packages for METcalcpy and METplotpy are listed in the pages below, respectively:
https://metcalcpy.readthedocs.io/en/develop/Users_Guide/installation.html
https://metplotpy.readthedocs.io/en/develop/Users_Guide/installation.html
 
It would be helpful for contributors to METplus wrappers, METcalcpy, and METplotpy, to continuously update the spreadsheet for less effort in the future and also for reference by others.
https://docs.google.com/spreadsheets/d/1Lf-yxZBXTTnBaqCOWUk-jdP9RpaLfil_s-KKH29CkKU/edit#gid=0

### Time Estimate ###
1 day of work.

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
ASAP

### Funding Source ###
Due to low time commitment, I'll plan to split.

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [x] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by jprestop**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>

Reviewed webpage where changes were added regarding updating the spreadsheet:

See External Dependency addition in section 8.4.4.2. Add Sphinx Documentation File at: 
https://metplus.readthedocs.io/en/feature_1374_python_packages/Contributors_Guide/add_use_case.html#document-new-use-case

See new section "Documenting the Addition of a New Python Package Requirement" under "8.5.4.1.3. Creating New Python Environments":
https://metplus.readthedocs.io/en/feature_1374_python_packages/Contributors_Guide/add_use_case.html#add-use-case-to-the-test-suite

Checked a few other minor modifications including changing master_metplus.py to run_metplus.py on the following pages:

- https://metplus.readthedocs.io/en/feature_1374_python_packages/generated/met_tool_wrapper/MODE/MODE_python_embedding.html#sphx-glr-generated-met-tool-wrapper-mode-mode-python-embedding-py
- https://metplus.readthedocs.io/en/feature_1374_python_packages/generated/model_applications/s2s/UserScript_fcstGFS_obsERA_Blocking.html#sphx-glr-generated-model-applications-s2s-userscript-fcstgfs-obsera-omi-py
- https://metplus.readthedocs.io/en/feature_1374_python_packages/generated/model_applications/s2s/UserScript_obsERA_obsOnly_Blocking.html#sphx-glr-generated-model-applications-s2s-userscript-fcstgfs-obsera-omi-py
- https://metplus.readthedocs.io/en/feature_1374_python_packages/generated/model_applications/s2s/UserScript_fcstGFS_obsERA_WeatherRegime.html#sphx-glr-generated-model-applications-s2s-userscript-fcstgfs-obsera-omi-py

Checked a few other minor modifications due to incorrect formatting in the run_metplus.py area:

- https://metplus.readthedocs.io/en/feature_1374_python_packages/generated/met_tool_wrapper/TCPairs/TCPairs_tropical.html#sphx-glr-generated-met-tool-wrapper-tcpairs-tcpairs-tropical-py
- https://metplus.readthedocs.io/en/feature_1374_python_packages/generated/model_applications/marine_and_cryosphere/GridStat_MODE_fcstIMS_obsNCEP_sea_ice.html#sphx-glr-generated-model-applications-marine-and-cryosphere-gridstat-mode-fcstims-obsncep-sea-ice-py
- https://metplus.readthedocs.io/en/feature_1374_python_packages/generated/model_applications/s2s/UserScript_obsPrecip_obsOnly_CrossSpectraPlot.html#sphx-glr-generated-model-applications-s2s-userscript-obsprecip-obsonly-crossspectraplot-py
- https://metplus.readthedocs.io/en/feature_1374_python_packages/generated/model_applications/s2s/UserScript_obsERA_obsOnly_Stratosphere.html#sphx-glr-generated-model-applications-s2s-userscript-obsera-obsonly-stratosphere-py

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Review the files changed for content, typos, grammar, and formatting.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[N/A]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **01/28/22**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Contributors_Guide/add_use_case.rst [hunk 1]
```diff
@@ -830,6 +841,14 @@ Please note that some commands in the scripts are specific to
 the Docker environment and may need to be rerun to successfully
 build the environment locally.
 
+**Documenting the Addition of a New Python Package Requirement**
```
Comment (georgemccabe) [inline docs/Contributors_Guide/add_use_case.rst]: I don't think this info belongs in this section. This section is specific to setting up the automated tests. I think the info listed above contains the same information and makes sense there, so you can remove this content.

**Review by georgemccabe — CHANGES_REQUESTED** [review]
I added a single comment to remove one of the new sections. The rest of the content looks good. Thanks for changing master_metplus.py to run_metplus.py where you saw it. I noticed that these example commands include the -c argument to run_metplus.py which is no longer needed, but a clean up of the use case conf files to remove redundant information should be done to resolve that, so it is not necessary in this PR.

Once the extra section is removed, I approve.

File: docs/Contributors_Guide/add_use_case.rst (hunk 1 above)
Comment (jprestop) [reply docs/Contributors_Guide/add_use_case.rst]: Thanks for letting me know! I removed it.

**Review by georgemccabe — APPROVED** [review]
Looks good to me! I approve.

**Pull request merged**